streaming/cleanup.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package streaming
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	dataflow "cloud.google.com/go/dataflow/apiv1beta3"
+	datastream "cloud.google.com/go/datastream/apiv1"
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+	"google.golang.org/api/iterator"
+)
+
+type ShardCleanupOptions struct {
+	Dataflow   bool
+	Datastream bool
+	Pubsub     bool
+	Monitoring bool
+}
+
+type JobCleanupOptions struct {
+	Monitoring bool
+}
+
+func InitiateJobCleanup(ctx context.Context, options JobCleanupOptions, jobExecutionData JobExecutionData, project string) error {
```

Comment:
It is unclear from the name how JobCleanup is different from ShardCleanup. 
ShardCleanup should be a subset of this functionality

Reply:
I understand the reason for the confusion. Restructured code to make `ShardCleanup` a subset of the `JobCleanup` functionality.